unison-cli/src/Unison/CommandLine/Main.hs

```diff
@@ -98,10 +99,32 @@ main ::
   Codebase IO Symbol Ann ->
   Maybe Server.BaseUrl ->
   UCMVersion ->
-  ((Branch IO, Path.Absolute) -> IO ()) ->
+  (Branch IO -> STM ()) ->
+  (Path.Absolute -> STM ()) ->
   IO ()
-main dir welcome initialPath (config, cancelConfig) initialInputs runtime sbRuntime codebase serverBaseUrl ucmVersion notifyChange = do
-  root <- Codebase.getRootBranch codebase
+main dir welcome initialPath (config, cancelConfig) initialInputs runtime sbRuntime codebase serverBaseUrl ucmVersion notifyBranchChange notifyPathChange = Ki.scoped \scope -> do
+  rootVar <- newEmptyTMVarIO
+  initialRootCausalHash <- Codebase.getRootCausalHash codebase
+  _ <- Ki.fork scope $ do
+    root <- Codebase.getRootBranch codebase
+    atomically $ do
+      -- Try putting the root, but if someone else as already written over the root, don't
```

Comment:
Who else might have written over the root?

Reply:
The CLI itself; not 100% if it's possible to do that right now, maybe if you `delete.namespace .` it might replace the root with the empty namespace, but I figure it's good to be future-proof here.